src/medley/core.cljc

```diff
@@ -357,6 +357,75 @@
   ([pred coll]
    (rest (drop-while (complement pred) coll))))
 
+(defn partition-after
+  "Returns a lazy sequence of partitions, splitting after `(pred item)` returns
+  true. Returns a transducer when no collection is provided."
+  ([pred]
+   (fn [rf]
+     (let [part #?(:clj (java.util.ArrayList.)
+                   :cljs (js/Array.))
+           split? (volatile! false)]
+       (fn
+         ([] (rf))
+         ([result]
+          (let [result (if #?(:clj (.isEmpty part) :cljs (zero? (alength part)))
+                         result
+                         (let [v (vec #?(:clj (.toArray part) :cljs (aclone part)))]
+                           #?(:clj (.clear part) :cljs (set! part -length 0))
+                           (unreduced (rf result v))))]
+            (rf result)))
+         ([result x]
+          (if @split?
+            (let [result (rf result (vec #?(:clj (.toArray part) :cljs (aclone part))))]
```

Comment:
I avoided doing this initially because I assumed it's (very slightly) less performant. I added an extra commit with `:inline` metadata to fix this issue in clj. I guess this could be a macro to benefit cljs too, but I really dislike not being able to use reader-conditionals in cljc macros, so didn't bother.

Reply:
Also added a commit renaming the -clear and -append fns to make the mutation clearer.